data_structures/binary_search_tree.cpp

```diff
@@ -0,0 +1,174 @@
+/**
+ * \file
+ * \brief A simple tree implementation using structured nodes
+ *
+ * \todo update code to use C++ STL library features and OO structure
+ * \warning This program is a poor implementation - C style - and does not
+ * utilize any of the C++ STL features.
+ */
+#include <iostream>
+
+struct node {
+    int val;
+    node *left;
+    node *right;
+};
+
+struct queue {
+    node *t[100];
+    int front;
+    int rear;
+};
+
+queue q;
+
+void enqueue(node *n) { q.t[q.rear++] = n; }
+
+node *dequeue() { return (q.t[q.front++]); }
+
+void Insert(node *n, int x) {
+    if (x < n->val) {
+        if (n->left == NULL) {
+            node *temp = new node;
+            temp->val = x;
+            temp->left = NULL;
+            temp->right = NULL;
+            n->left = temp;
+        } else {
+            Insert(n->left, x);
+        }
+    } else {
+        if (n->right == NULL) {
+            node *temp = new node;
+            temp->val = x;
+            temp->left = NULL;
+            temp->right = NULL;
+            n->left = temp;
+        } else {
+            Insert(n->right, x);
+        }
+    }
+}
+
+int findMaxInLeftST(node *n) {
+    while (n->right != NULL) {
+        n = n->right;
+    }
+    return n->val;
+}
+
+void Remove(node *p, node *n, int x) {
+    if (n->val == x) {
+        if (n->right == NULL && n->left == NULL) {
+            if (x < p->val) {
+                p->right = NULL;
+            } else {
+                p->left = NULL;
+            }
+        } else if (n->right == NULL) {
+            if (x < p->val) {
+                p->right = n->left;
+            } else {
+                p->left = n->left;
+            }
+        } else if (n->left == NULL) {
+            if (x < p->val) {
+                p->right = n->right;
+            } else {
+                p->left = n->right;
+            }
+        } else {
+            int y = findMaxInLeftST(n->left);
+            n->val = y;
+            Remove(n, n->right, y);
+        }
+    } else if (x < n->val) {
+        Remove(n, n->left, x);
+    } else {
+        Remove(n, n->right, x);
+    }
+}
+
+void BFT(node *n) {
+    if (n != NULL) {
+        std::cout << n->val << "  ";
+        enqueue(n->left);
+        enqueue(n->right);
+        BFT(dequeue());
+    }
+}
+
+void Pre(node *n) {
```

Comment:
why we are not naming these functions as `preOrder`, `postOrder` and `inOrder` instead of just `Pre` , `Post`, `In`, i think  `preOrder`, `postOrder` and `inOrder` looks more readable isn't it?

Reply:
You are correct. Please check the file in the main repo. Such files were accepted and hence, it is a Herculean effort to fix. 
My goal, is stated above. Such camel fixes can be made in a later pull request.

For one man to fix everything, it is not a simple task :)

Reply:
yes you are right, no probelm we will fix these small issues later on